api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineRunManagerInstanceTypePriceTypeAndCloudRegionTest.java

```diff
@@ -0,0 +1,428 @@
+import com.epam.pipeline.manager.docker.ToolVersionManager;
+import com.epam.pipeline.manager.execution.PipelineLauncher;
+import com.epam.pipeline.manager.preference.PreferenceManager;
+import com.epam.pipeline.manager.region.CloudRegionManager;
+import com.epam.pipeline.manager.security.AuthManager;
+import com.epam.pipeline.manager.security.CheckPermissionHelper;
+import com.epam.pipeline.manager.security.run.RunPermissionManager;
+import io.reactivex.subjects.BehaviorSubject;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+import java.util.HashMap;
+import java.util.Optional;
+
+import static com.epam.pipeline.common.MessageConstants.ERROR_INSTANCE_TYPE_IS_NOT_ALLOWED;
+import static com.epam.pipeline.common.MessageConstants.ERROR_PRICE_TYPE_IS_NOT_ALLOWED;
+import static com.epam.pipeline.common.MessageConstants.ERROR_RUN_CLOUD_REGION_NOT_ALLOWED;
+import static com.epam.pipeline.common.MessageConstants.ERROR_TOOL_CLOUD_REGION_NOT_ALLOWED;
+import static com.epam.pipeline.entity.contextual.ContextualPreferenceLevel.TOOL;
+import static com.epam.pipeline.manager.preference.SystemPreferences.CLUSTER_DOCKER_EXTRA_MULTI;
+import static com.epam.pipeline.manager.preference.SystemPreferences.CLUSTER_INSTANCE_HDD_EXTRA_MULTI;
+import static com.epam.pipeline.manager.preference.SystemPreferences.CLUSTER_SPOT;
+import static com.epam.pipeline.manager.preference.SystemPreferences.COMMIT_TIMEOUT;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ID;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ID_2;
+import static com.epam.pipeline.test.creator.CommonCreatorConstants.ID_3;
+import static com.epam.pipeline.test.creator.configuration.ConfigurationCreatorUtils.getPipelineConfiguration;
+import static com.epam.pipeline.test.creator.docker.DockerCreatorUtils.getTool;
+import static com.epam.pipeline.test.creator.pipeline.PipelineCreatorUtils.getPipelineRunWithInstance;
+import static com.epam.pipeline.test.creator.region.RegionCreatorUtils.getDefaultAwsRegion;
+import static com.epam.pipeline.test.creator.region.RegionCreatorUtils.getNonDefaultAwsRegion;
+import static com.epam.pipeline.util.CustomAssertions.assertThrows;
+import static java.lang.Integer.parseInt;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyInt;
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@SuppressWarnings("PMD.UnusedPrivateField")
+public class PipelineRunManagerInstanceTypePriceTypeAndCloudRegionTest {
```

Comment:
What do you think about renaming the test class to `PipelineRunManagerInstanceSettingsTest`? It seems like instance type, price type and cloud region can be referred as instance settings.

Reply:
I have already renamed it to "PipelineRunManagerLaunchTest" in this branch: https://github.com/epam/cloud-pipeline/pull/1797

according to this issue and Maria's comment : https://github.com/epam/cloud-pipeline/issues/1776#issuecomment-776663434

So I will rename it to PipelineRunManagerLaunchTest there in order to avoid merge conflict.